### PR TITLE
Update Registry README and some example files for x-translator and x-trapi

### PR DESCRIPTION
**Pull request by RichardBruskiewich**

### CODE REVIEW

**Comment by newgene** [comment]
@RichardBruskiewich can you please change the referenced URL from your fork to this original repo?

File: README.md [hunk 1]
```diff
@@ -6,26 +6,63 @@ The Translator-specific metadata extensions used in the SmartAPI registry
...
+  - name: translator    ## required translator tag
+  - name: biothings
+  info:
+    x-trapi:      ## required x-translator extension
+      version: "1.2.0"
+      operations:
+      - OperationLookup
+      test_data_location: "https://github.com/biothings/biothings.api/blob/master/test_data"  # just a fictional example link
+```
+This same example in JSON and other examples of valid and invalid instances (in JSON) are in [this example file](https://github.com/STARInformatics/translator_extensions/blob/main/x-trapi/smartapi_x-trapi_examples.txt).
```
[the remark this answers is not in the document]
Comment (RichardBruskiewich) [reply README.md]: Hi @newgene, so very sorry that I misunderstood what you were asking for here. Fixed. I hope that the PR is in order now.

File: README.md [hunk 2]
```diff
@@ -1,31 +1,69 @@
...
+
 The Translator-specific metadata extensions used in the SmartAPI registry
 
 ## x-translator
 #### Status
 This extension's schema has been written, and registry files currently contain this extension.
 
 #### Description
-This extension is inside the OpenAPI info object and contains basic API-level metadata. Currently...
+This extension is inside the OpenAPI info object and contains basic API-level metadata. Currently, there are two (required) properties: component (KP, ARA, ARS, or Utility) and team (one or more Translator team names). Current tags include:
```
Comment (colleenXu) [inline README.md:10]: Change "tags" to properties?

File: README.md [hunk 3]
```diff
@@ -1,31 +1,69 @@
...
+
+- **infores:** Information Resource CURIE identifying the Translator Resource
```
Comment (colleenXu) [inline README.md:12]: Maybe link the spreadsheet that has the infores stuff, or some details on how to add / modify / delete an infores?

File: README.md [hunk 4]
```diff
@@ -1,31 +1,69 @@
...
+- **component:** One of  "KP", "ARA", "ARS", "Utility"
+- **team**: List of contributing Translator teams (see [schema file](https://github.com/NCATSTranslator/translator_extensions/blob/main/x-translator/smartapi_x-translator_schema.json) for the full list)
+- **biolink-version:**  One SemVer-formatted version number (e.g. '1.8.2', '2.2.16', etc.)
```
Comment (colleenXu) [inline README.md:15]: Give more detail. "This is the biolink-model version that the Translator Resource is currently compliant with / was built to...and this is where you go to look at biolink-model and its versions / releases."

File: README.md [hunk 5]
```diff
@@ -1,31 +1,69 @@
...
   info:
     x-translator:      ## required x-translator extension
+      infores: "infores:biolink-api"
       component: KP
       team:
       - "Service Provider"
+      biolink-version: "2.2.16"
 ```
-This same example in JSON and other examples of valid and invalid instances (in JSON) are in [this file](https://github.com/NCATSTranslator/translator_extensions/blob/main/x-translator/smartapi_x-translator_examples.txt). 
+This same example in JSON and other examples of valid and invalid instances (in JSON) are in [this example file](https://github.com/NCATSTranslator/translator_extensions/blob/main/x-translator/smartapi_x-translator_examples.txt).
```
Comment (colleenXu) [inline README.md:37]: Then that file for x-translator examples needs updating...

File: README.md [hunk 6]
```diff
@@ -1,31 +1,69 @@
...
+
+#### Status
+
+This extension's schema has been started, and registry files currently contain this extension.
+
+#### Description
+
+This extension is inside the OpenAPI info object and contains basic API-level metadata. Currently, there are three (required) properties: version, operations and test_data_location, plus other optional ones:
+
+- **version:** One TRAPI version SemVer formatted number (e.g. '1.1', '1.2', '1.3')
```
Comment (colleenXu) [inline README.md:49]: Give more info? "States the TRAPI standard version that the Translator Resource is compliant with / built to. The TRAPI standard and its versions can be viewed here".

File: README.md [hunk 7]
```diff
@@ -1,31 +1,69 @@
...
-This extension's schema has not been written / registry files do not contain this extension. It will likely contain API- and operation-level metadata that TRAPI expects.
\ No newline at end of file
+
+#### Status
+
+This extension's schema has been started, and registry files currently contain this extension.
+
+#### Description
+
+This extension is inside the OpenAPI info object and contains basic API-level metadata. Currently, there are three (required) properties: version, operations and test_data_location, plus other optional ones:
```
Comment (colleenXu) [inline README.md:47]: Note that in the schema, currently only version is required. Operations and test_data_location actually aren't. there also isn't a enum to constrict the possible string values of operations.

Comment (RichardBruskiewich) [reply README.md:47]: @putmantime may wish to comment here: I think the Translator expectation will soon be to make `test_data_location` mandatory within TRAPI (`x-trapi` tagged) entries?

I can't comment on the `operations` property - that would perhaps be the remit of @dkoslicki or @cbizon.

**Comment by RichardBruskiewich** [comment]
@colleenXu, please see my last commit. Aside from the issue of 'mandatory' properties for x-trapi (which I delegate to others above), I've tried to fix the README and sample files a little bit to resolve your concerns above.

cc: @putmantime, @dkoslicki and @cbizon for additional feedback.

**Comment by cbizon** [comment]
> @colleenXu, please see my last commit. Aside from the issue of 'mandatory' properties for x-trapi (which I delegate to others above), I've tried to fix the README and sample files a little bit to resolve your concerns above.
> 
> cc: @putmantime, @dkoslicki and @cbizon for additional feedback.

FWIW, I would support making both `test_data_location` and `operations` mandatory.

**Comment by colleenXu** [comment]
@RichardBruskiewich I've reviewed the changes and resolved several comments. I've left two open:
* https://github.com/NCATSTranslator/translator_extensions/pull/16#discussion_r881212176: just involves adding a bit to the README 
* https://github.com/NCATSTranslator/translator_extensions/pull/16#discussion_r881214323: this involves edits to the schema itself. It may include enumerating the operations....

**Comment by RichardBruskiewich** [comment]
@colleenXu, I'm not that familiar with the finer points of JSON schemata, but I made the  both `test_data_location` and `operations mandatory` 'required' in the `x-trapi` extension schema. I hope this suffices. I added a sentence dereferencing the Biolink Model (and its releases). On the issue of the operations, once again, I'm agnostic. If you don't wish to merge this PR yet pending some resolution of that issue, perhaps coordinate things further with @putmantime, @dkoslicki and @cbizon.

**Comment by colleenXu** [comment]
@RichardBruskiewich Your changes from https://github.com/STARInformatics/translator_extensions/commit/5ea1951ed5aaf81b183d382eeacb36a06728dfbd look okay. 

However, that commit doesn't seem to be on the main branch of your fork. And it is the main branch of your fork that is being used in this PR...

**Comment by RichardBruskiewich** [comment]
> Noting what I think is unresolved: a list of operations when making the operations field mandatory

@dkoslicki , @edeutsch - can you comment?

This issue was originally opened to updated Translator SmartAPI Registry document mainly with respect to TRAPI and operations standards in your remit.

Please kindly advise Colleen about this.

**Comment by colleenXu** [comment]
Andrew's suggestion is to put the schema changes from this PR into a new PR instead. Once the schema changes are approved / pushed to prod, it'll start being used to validate smartapi registrations....and this will break / raise errors with the existing registrations. So we only want to push once people are ready for that (or have updated their registrations). 

The title of this PR suggests non-breaking changes to documentation, which seems like an easy/fast/non-controversial change.

**Comment by RichardBruskiewich** [comment]
> Andrew's suggestion is to put the schema changes from this PR into a new PR instead. Once the schema changes are approved / pushed to prod, it'll start being used to validate smartapi registrations....and this will break / raise errors with the existing registrations. So we only want to push once people are ready for that (or have updated their registrations).
> 
> The title of this PR suggests non-breaking changes to documentation, which seems like an easy/fast/non-controversial change.

You folks own the repo... go ahead and do that

**Comment by edeutsch** [comment]
So if I understand this all correctly, there is one new thing: test_data_location
and the rest is all just documentation stuff? is that right?

I agree that it would be nice to separate that which is just documentation updates that don't affect implementers much, and things that are new that will require implementers to act on

**Comment by colleenXu** [comment]
I'm going to close this PR and open another one from a branch in this repo. That should make it easier for all parties to edit. The README and some examples have already been updated on main, based on the stuff that was in this PR.

**Comment by RichardBruskiewich** [comment]
sounds good

**Pull request closed without merging**